src/util/hash_util.rs

```diff
@@ -66,6 +73,42 @@ fn murmur_hash2_64a<T: AsBytes>(data: &T, seed: u64) -> u64 {
   h
 }
 
+/// CRC32 hash implementation using SSE4 instructions. Borrowed from Impala.
+#[cfg(target_feature="sse4.2")]
+pub fn crc32_hash<T: AsBytes>(data: &T, seed: u32) -> u32 {
+  let bytes: &[u8] = data.as_bytes();
+  let u32_num_bytes = ::std::mem::size_of::<u32>();
+  let mut num_bytes = bytes.len();
+  let num_words = num_bytes / u32_num_bytes;
+  num_bytes %= u32_num_bytes;
+
+  let bytes_u32: &[u32] = unsafe {
```

Comment:
Is it necessary to use `unsafe` block with from_raw_parts? Would it be reasonable to consider any other way of constructing `bytes_u32`?

Reply:
Either use `unsafe` or we have to move the bits around and construct u32 from 4 u8s. I think it is "safe" to use the `unsafe` block here, since we know the length of the slice.

Reply:
Yeah, you are right!